File: Autres parcours/Mathématiques/Quelques outils mathématiques/4 - Approvisionnement de la ruche/program.py
from math import *
import sys


def saisie():
    valeur = int(sys.stdin.readline().strip())
    return valeur


def main():
    x_ruche = saisie()
    y_ruche = saisie()
    nbfleurs = saisie()
    fleurs_accessible = 0
    for i in range(nbfleurs):
        x = saisie()
        y = saisie()
        distance = sqrt((x - x_ruche)**2 + (y - y_ruche)**2)
        if distance < 1000:
            fleurs_accessible += 1
    print(fleurs_accessible)


main()

File: Autres parcours/Mathématiques/Quelques outils mathématiques/4 - Approvisionnement de la ruche/program.c
#include <stdio.h>
#include <math.h>

int main()
{
   int x_ruche, y_ruche, nbFleurs;
   
   scanf("%d%d%d", &x_ruche, &y_ruche, &nbFleurs);
   int fleurs_accessible = 0;
   for (int i = 1; i <= nbFleurs; i++)
   {
      int x, y;
      float distance;
      
      scanf("%d%d", &x, &y);
      distance = sqrt(pow(x_ruche - x, 2) + pow(y_ruche - y, 2));
      
      if (distance < 1000.0)
      {
         fleurs_accessible++;
      }
   }
   printf("%d", fleurs_accessible);
   return 0;
}
